src/csharp/Grpc.Core.Tests/CompressionTest.cs

```diff
@@ -118,5 +119,30 @@ public void WriteOptions_Unary()
 
             await call.ResponseStream.ToListAsync();
         }
+
+        [Test]
+        public void CanReadCompressedMessages()
+        {
+            var compressionMetadata = new Metadata
+            {
+                { new Metadata.Entry("grpc-internal-encoding-request", "gzip") }
```

Comment:
Actually, would it be possible to use `GRPC_COMPRESSION_REQUEST_ALGORITHM_MD_KEY` from https://github.com/grpc/grpc/blob/master/include/grpc/impl/codegen/compression_types.h#L46?